go-controller/pkg/ovn/gress_policy.go

```diff
@@ -51,25 +51,64 @@ type portPolicy struct {
 	endPort  int32
 }
 
-func (pp *portPolicy) getL4Match() (string, error) {
+// for a given ingress/egress rule, captures all the provided port ranges and
+// individual ports
+type gresssPolicyPorts struct {
+	portList  []string // list of provided ports as string
+	portRange []string // list of provided port range in OVN ACL format
+}
+
+func (gp *gressPolicy) getProtocolPortsMap() map[string]*gresssPolicyPorts {
 	var supportedProtocols = []string{TCP, UDP, SCTP}
-	var foundProtocol string
-	for _, protocol := range supportedProtocols {
-		if protocol == pp.protocol {
-			foundProtocol = strings.ToLower(pp.protocol)
-			break
+	gressProtoPortsMap := make(map[string]*gresssPolicyPorts)
+	for _, pp := range gp.portPolicies {
+		var found bool
+		for _, protocol := range supportedProtocols {
+			if protocol == pp.protocol {
+				found = true
+				break
+			}
+		}
+		if !found {
+			klog.Warningf("Unknown protocol %v, while processing network policy %s/%s",
+				pp.protocol, gp.policyNamespace, gp.policyName)
+			continue
+		}
+		protocol := strings.ToLower(pp.protocol)
+		gpp, ok := gressProtoPortsMap[protocol]
+		if !ok {
+			gpp = &gresssPolicyPorts{portList: []string{}, portRange: []string{}}
+			gressProtoPortsMap[protocol] = gpp
+		}
+		if pp.endPort != 0 && pp.endPort != pp.port {
```

Comment:
Should we check that `pp.port < pp.endPort` or is this done elsewhere in the code?

Reply:
@girishmg is mine concern here valid?

Reply:
@ricky-rav that check is done by the k8s controller natively.. so, no need to check for that condition explicitly...